src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/model/CaseRecordFactory.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model;
+
+import com.google.common.collect.ImmutableMap;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.caserecord.ExceptionRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.caserecord.SupplementaryEvidenceRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+
+import java.util.EnumMap;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.CaseRecord.Construct;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.model.CaseRecord.Record;
+
+public final class CaseRecordFactory {
+
+    private static final Map<Record, Construct> AVAILABLE_MODELS;
+
+    static {
+        AVAILABLE_MODELS = new EnumMap<>(ImmutableMap.of(
+            Record.EXCEPTION_RECORD, ExceptionRecord::new,
+            Record.SUPPLEMENTARY_EVIDENCE, SupplementaryEvidenceRecord::new
+        ));
```

Comment:
this is now a (nicely formatted) one liner and could be moved to field declaration I think
ie:
```
private static final Map<Record, Construct> AVAILABLE_MODELS = new ...
```